### ISSUE TITLE
Inconsistent padding on Dua Detail List Card

### ISSUE THREAD

**Issue by khalid-hussain** [khalid-hussain/HisnulMuslim#7]
The problem can be seen clearly from the screenshot. The 'reference' text view has both a bottom and top padding whereas the 'translation' does not. This is probably because of it being not a regular text view but rather an older implementation of the [TextJustify-Android]() library.

Possible culprits:
1. [dua_detail_item_card.xml](https://github.com/khalid-hussain/HisnulMuslim/blob/master/app/src/main/res/layout/dua_detail_item_card.xml)
2. [TextViewEx.java](https://github.com/khalid-hussain/HisnulMuslim/blob/master/app/src/main/java/classes/TextViewEx.java)

The problem may also solve itself if [Issue #6](https://github.com/khalid-hussain/HisnulMuslim/issues/6) is solved.

![Image](http://s10.postimg.org/7blszubah/device_2015_01_10_044304.png)

**Comment by mdali92** [comment]
Go to dua_detail_item_card.xml and do the following edit :

Basically for txtDuaReference, remove the top padding property and change bottom padding to 5dp, like this :

<TextView
                android:id="@+id/txtDuaReference"
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                android:paddingBottom="5dp"
                android:text="Dua Reference"
                android:textSize="14sp" />

It should look better now.

![screenshot_2015-01-11-11-03-22](https://cloud.githubusercontent.com/assets/7617302/5694196/64848468-9982-11e4-9d21-bab2da2f1dea.png)

**Comment by khalid-hussain** [comment]
Actually I wanted the opposite. I want both the 'translation' and 'reference' to have 10dp padding on the top and bottom.

**Comment by mdali92** [comment]
Alright, here is the fix. There is a small trick. Your English translation is using the class TextViewEx and not the standard TextView. This is giving issues when you are adding padding above and below the english text. However, the reference text uses the standard TextView and padding is not an issue there. Therefore, the trick is to add 10dp margin to the divider grid above and below the English text... like this :

```
           <View
            android:layout_width="match_parent"
            android:layout_height="1dp"
            android:layout_marginLeft="1dp"
            android:layout_marginRight="1dp"
            android:layout_marginBottom="10dp"
            android:background="@color/card_divider_gray" />

        <classes.TextViewEx
            android:id="@+id/txtDuaTranslation"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:gravity="center_vertical"
            android:text="English Text"
            android:textSize="14sp" />

        <View
            android:layout_width="match_parent"
            android:layout_height="1dp"
            android:layout_marginLeft="1dp"
            android:layout_marginRight="1dp"
            android:layout_marginTop="10dp"
            android:background="@color/card_divider_gray" />

        <TextView
            android:id="@+id/txtDuaReference"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:paddingBottom="10dp"
            android:paddingTop="10dp"
            android:text="Dua Reference"
            android:textSize="14sp" />
```

Now the result looks like this :  

![screenshot_2015-01-11-15-55-03](https://cloud.githubusercontent.com/assets/7617302/5694647/b74d94d6-99ab-11e4-9a7b-cf4f1d8b0615.png)

**Comment by khalid-hussain** [comment]
Following your solution, this is what has been achieved:
![](http://s25.postimg.org/7yjk6kuxb/Untitled2.png)

I still don't understand why there is this extra spacing. I have a feeling that to creating an efficient solution involves editing [TextViewEx](https://github.com/khalid-hussain/HisnulMuslim/blob/master/app/src/main/java/classes/TextViewEx.java).

JazakAllahu khayran for the effort.

**Comment by mdali92** [comment]
That's the 10dp padding. I thought you said you wanted 10dp padding at the top and bottom of translation and references?

**Comment by khalid-hussain** [comment]
I've used the same padding settings for both the regular textView and the textView ex. The 10dp for the 'translation' is seen as the pink bars. In the image to the right, there is some extra spacing between the textView ex and the spacing/padding for the divider View. We need to find out where it is coming from and remove it.

**Comment by khalid-hussain** [comment]
This has been fixed due to commit bc9f510162262612547750680501aebbab4aa0af

Preview:
![](http://s25.postimg.org/ed9uv53tb/fixed_screen.png)

**Comment by khalid-hussain** [comment]
Due to the update of the Text-Justify library to 2.0.4, this is back.
